### PR TITLE
Add More Event Logs for Analytics 1.0

### PR DESCRIPTION
**Pull request by hanywang2**
## Summary

Add logging for the following events when a user 
- Clicks on the CTA
- Clicks on the navigation item
- Copies a code snippet (from the codeblock)
- Makes an API call
- Scrolls to the bottom of the page

## Test Plan

Make sure to help test this one thoroughly. To do so, add the following code snippets to L135-136 of `AnalyticsMediator.ts`

```javascript
console.log(eventName);
console.log(eventProperties);
```

CleanShot 2023-02-05 at 16 42 47@2x

Then click around for the following events and you should see the above events get triggered properly.

**Note**: Make sure to double check that things aren't getting double logged

### CODE REVIEW

**Comment by hahnbeelee** [comment]
Thank you for the test plan uwu :3

**Review by hahnbeelee — CHANGES_REQUESTED** [review]
I scrolled to the bottom of the page and I didn't get a console log on http://mintlify.localhost:3000/quickstart

image

**Review by hahnbeelee — CHANGES_REQUESTED** [review]
- [x] Clicks on the CTA
- [x] Header nav item click
- [ ] Clicks on the navigation item - nothing is console logging for me
- [x] Copies a code snippet (from the codeblock)
- [x] Makes an API call
- [x] Accordion open + close
- [ ] Scrolls to the bottom of the page
^ Didn't work on http://mintlify.localhost:3000/quickstart
Logged scroll_to_bottom twice on http://mintlify.localhost:3000/components/accordion
image

**Comment by hahnbeelee** [comment]
In the future I think we should architect this in a way so that we can define the inputs for the types for the different event handlers.

**Comment by hahnbeelee** [comment]
Some ideas for events to track:
- Clicking on an anchor
- Might be worth to figure out how to take the title of code blocks as inputs? Not sure how difficult implementing this would be.

File: client/src/ui/Header.tsx [hunk 1]
```diff
@@ -180,6 +192,7 @@ function TopBarCtaButton({ button }: { button: TopbarCta }) {
           href={button.url ?? '/'}
           target="_blank"
           className="group px-4 py-1.5 relative inline-flex items-center rounded-full shadow-sm text-sm font-medium"
+          onClick={() => trackCtaClick({ name: button.name, url: button.url })}
```
Comment (tobias-mintlify) [inline client/src/ui/Header.tsx:195]: Should we pass a `type` event property here? Maybe `type="button"`?

File: client/src/ui/MDXContentController/MDXContentController.tsx [hunk 2]
```diff
@@ -75,6 +77,31 @@ export function MDXContentController({
...
+  // Log event when scroll to bottom
+  useEffect(() => {
+    if (checkIfAtBottomOfPage()) {
+      trackScrollToBottom({ path: Router.asPath });
+      return;
+    }
+
+    const onScroll = () => {
+      if (checkIfAtBottomOfPage()) {
+        trackScrollToBottom({ path: Router.asPath });
```
Comment (tobias-mintlify) [inline client/src/ui/MDXContentController/MDXContentController.tsx:97]: `asPath` will include other things in the URL like the header ID when permalinked: `/my-page#header-id`. You may want to use `useCurrentPath()` and pass that to useEffect. Passing `useCurrentPath` as a dependency might also fix a potential bug I mention in the next comment.

File: client/src/ui/MDXContentController/MDXContentController.tsx [hunk 3]
```diff
@@ -75,6 +77,31 @@ export function MDXContentController({
...
+        window.removeEventListener('scroll', onScroll);
+      }
+    };
+    window.addEventListener('scroll', onScroll);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```
Comment (tobias-mintlify) [inline client/src/ui/MDXContentController/MDXContentController.tsx:103]: Please test this carefully. I'm not sure the event will fire multiple times if you scroll to the bottom of multiple pages. The listener might be removed after the first time and never placed again.

File: client/src/ui/search/Search.tsx [hunk 4]
```diff
@@ -305,7 +310,13 @@ export function SearchProvider({
       .toLowerCase()
       .replaceAll(' ', '-')
       .replaceAll(/[^a-zA-Z0-9-_#]/g, '');
-    router.push(`/${hit.slug}${sectionSlug}`);
+    router.push(`${path}${sectionSlug}`);
```
Comment (tobias-mintlify) [inline client/src/ui/search/Search.tsx:313]: Please avoid using `path` as a variable name. We use that in too many places. It's okay to merge this for now though.

**Pull request closed without merging**